source/utilities.cc

```diff
@@ -2532,6 +2532,21 @@ namespace aspect
     }
 
 
+    template <int dim>
+    SymmetricTensor<2,dim>
+    nth_basis_for_symmetric_tensors (const unsigned int k)
+    {
+      // somehow I need this, because otherwise it complains that I am passing 3 arguments and it only takes 2...
+      const bool boolian = k < SymmetricTensor<2,dim>::n_independent_components;
+      Assert(boolian, ExcMessage("The component is larger then the amount of independent components in the matrix.") );
```

Comment:
You can write this as
```
  Assert((k < SymmetricTensor<2,dim>::n_independent_components), 
             ExcMessage("The component is larger then the amount of independent components in the matrix.") );
```
This has the advantage that the violated condition appears in the screen output.

(Unrelated: The correct spelling is "boolean" because it is named after Mr. Boole.)